endpoints/authentication/src/main.cpp

```diff
@@ -163,13 +186,23 @@ namespace irods::http::handler
 		return {std::move(username), std::move(password)};
 	}
 
+	auto is_oidc_running_as_client() -> bool
+	{
+		static const auto oidc_stanza_exists = irods::http::globals::configuration().contains(
+			nlohmann::json::json_pointer{"/http_server/authentication/openid_connect"});
+
+		if (oidc_stanza_exists) {
+			static const auto is_client{irods::http::globals::oidc_configuration().at("run_as_protected_resource").get<bool>()};
+			return is_client;
```

Comment:
okay, so this *is* an either/or situation...   so an alternative universe could have a 'mode' set in the configuration with an enum of ... ['client', 'protected_resource'].

Reply:
or even make the oidc stanza required... but with a 'mode' that could be one of ['none', 'client', 'protected_resource'].

could provide flexibility for future good ideas.

Reply:
Going to point this one here: https://github.com/irods/irods_client_http_api/pull/252#discussion_r1498535591

It'll be closed when the linked on is.